sdk/communication/azure-communication-administration/azure/communication/administration/_polling.py

```diff
@@ -0,0 +1,95 @@
+# pylint: disable=W0231
+# -------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for
+# license information.
+# --------------------------------------------------------------------------
+import base64
+import time
+from typing import Union
+from functools import partial
+import pickle
+
+from azure.core.polling import (
+    PollingMethod
+)
+from ._phonenumber._generated.models import (
+    PhoneNumberSearch,
+    PhoneNumberRelease
+)
+
+class PhoneNumberBasePolling(PollingMethod):
+    """ABC class for reserve/purchase/release phone number related polling.
+    """
+    def __init__(self, is_terminated, interval=5):
+        self._response = None
+        self._client = None
+        self._query_status = None
+        self._is_terminated = is_terminated
+        self._polling_interval = interval
+
+    def _update_status(self):
+        # type: () -> None
+        if self._query_status is None:
+            raise Exception("this poller has not been initialized")
+        self._response = self._query_status()  # pylint: disable=E1102
+
+    def initialize(self, client, initial_response, deserialization_callback):
+        # type: (Any, Any, Callable) -> None
+        self._client = client
+        self._response = initial_response
+
+    def run(self):
+        # type: () -> None
+        while not self.finished():
+            self._update_status()
+            if not self.finished():
+                time.sleep(self._polling_interval)
+
+    def finished(self):
+        # type: () -> bool
+        if self._response.status is None:
+            return False
+        return self._is_terminated(self._response.status)
+
+    def resource(self):
+        # type: () -> Union[PhoneNumberSearch, PhoneNumberRelease]
+        if not self.finished():
+            return None
+        return self._response
+
+    def status(self):
+        # type: () -> str
+        return self._response.status
+
+    def get_continuation_token(self):
+        # type() -> str
+        return base64.b64encode(pickle.dumps(self._response)).decode('ascii')
+
+    @classmethod
+    def from_continuation_token(cls, continuation_token, **kwargs):
```

Comment:
If client is required, why not put it as a positional argument?

Reply:
Thanks for the review. This is to override from_continuation_token method from base class [PollingMethod](https://github.com/Azure/azure-sdk-for-python/blob/a6f20b9cb5ee941b8e540d57e419528065f06f62/sdk/core/azure-core/azure/core/polling/_poller.py#L74)

Reply:
I think as long as they have same name, it overrides the method from base class.

Reply:
Thanks! Modified.